src/csv/index.js

```diff
@@ -200,21 +201,38 @@ const adapterFactories = {
   }),
 };
 
-function classifyFile(firstRow) {
-  if (arrayEqual(firstRow, boaCreditHeader)) return 'boaCredit';
-  if (arrayEqual(firstRow, boaBankHeader)) return 'boaBank';
-  if (arrayEqual(firstRow, venmoHeader)) return 'venmo';
-  if (arrayEqual(firstRow, cashAppHeader)) return 'cashApp';
-  if (arrayEqual(firstRow, capitalOneHeader)) return 'capitalOne';
-  if (firstRow[0] && firstRow[0].startsWith('Account Name : ')) return 'delta';
+function checkHeaderPatterns(possibleHeaderRow) {
+  if (arrayEqual(possibleHeaderRow, boaCreditHeader)) return 'boaCredit';
+  if (arrayEqual(possibleHeaderRow, boaBankHeader)) return 'boaBank';
+  if (arrayEqual(possibleHeaderRow, venmoHeader)) return 'venmo';
+  if (arrayEqual(possibleHeaderRow, cashAppHeader)) return 'cashApp';
+  if (arrayEqual(possibleHeaderRow, capitalOneHeader)) return 'capitalOne';
+  if (
+    possibleHeaderRow[0] &&
+    possibleHeaderRow[0].startsWith('Account Name : ')
+  )
+    return 'delta';
 
   return '';
 }
 
 function normalizeDate(date) {
   const d = new Date(Date.parse(date));
+  if (
+    Number.isNaN(d.getMonth()) ||
+    Number.isNaN(d.getDate()) ||
+    Number.isNaN(d.getYear())
+  )
+    throw new Error(`failed to normalize date string: "${date}"`);
```

Comment:
more explicit error propagation, instead of letting bad data cause Postgres failures